packages/boot/test/unit/booters/repository.booter.unit.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect, TestSandbox, sinon} from '@loopback/testlab';
+import {Application} from '@loopback/core';
+import {RepositoryMixin} from '@loopback/repository';
+import {RepositoryBooter, RepositoryDefaults} from '../../../index';
+import {resolve} from 'path';
+
+describe('repository booter unit tests', () => {
+  const SANDBOX_PATH = resolve(__dirname, '../../../.sandbox');
+  const sandbox = new TestSandbox(SANDBOX_PATH);
+
+  const REPOSITORIES_PREFIX = 'repositories';
+  const REPOSITORIES_TAG = 'repository';
+
+  class RepoApp extends RepositoryMixin(Application) {}
+
+  let app: RepoApp;
+  let stub: sinon.SinonStub;
+
+  beforeEach(() => sandbox.reset());
+  beforeEach(getApp);
+  before(createStub);
+  after(restoreStub);
+
+  it('gives a warning if called on a app without RepositoryMixin', async () => {
+    const normalApp = new Application();
+    // tslint:disable-next-line:no-unused-expression
+    new RepositoryBooter(normalApp, SANDBOX_PATH);
+    sinon.assert.calledOnce(stub);
+    sinon.assert.calledWith(
+      stub,
+      'app.repository() function is needed for RepositoryBooter. You can add it to your Application using RepositoryMixin from @loopback/repository.',
+    );
+  });
+
+  it(`uses RepositoryDefaults for 'options' if none are give`, () => {
+    const booterInst = new RepositoryBooter(app, SANDBOX_PATH);
+    expect(booterInst.options).to.deepEqual(RepositoryDefaults);
+  });
+
+  it('overrides defaults with provided options and uses defaults for the rest', () => {
+    const options = {
+      dirs: ['test'],
+      extensions: ['.ext1'],
+    };
+    const expected = Object.assign({}, options, {
+      nested: RepositoryDefaults.nested,
+    });
+
+    const booterInst = new RepositoryBooter(app, SANDBOX_PATH, options);
+    expect(booterInst.options).to.deepEqual(expected);
+  });
+
+  it('binds repositories during the load phase', async () => {
+    const expected = [
+      `${REPOSITORIES_PREFIX}.ControllerOne`,
```

Comment:
I may missed some chat in old PR: wouldn't it be `${REPOSITORIES_PREFIX}.RepositoryOne`?
I think the name `ControllerOne` come from https://github.com/strongloop/loopback-next/blob/567ae7be9ec661424ac67eb406c8894b0f8999f0/packages/boot/test/fixtures/multiple.artifact.ts#L8
Does that mean we just need a class for test, it acts as any possible artifact?

Reply:
For the purpose of a test, I'm using a Class as a `Repository` is a class. As a test I think this validates the fact that a Class in the appropriate file got bound to the Application. 

I did this to keep the number of fixtures to a minimum is all.

Reply:
I think if that's the case, there should be a comment above the constant declaration that clarifies why we're binding a controller class to a repository class.

Reply:
👍 thank you I see, as Kyu says some comment or rename sounds better, while not a blocker for this PR IMO :), just some nitpick.

Reply:
I have renamed the class names to be more generic, `ArtifactOne` and `ArtifactTwo` ... which makes sense since they live in `multiple.artifact.ts`. That should create less confusion and allow us to test other Booters which just need a class using the same file.